### ISSUE TITLE
Access Token seems to run out after a short time

### ISSUE THREAD

**Issue by acucciniello** [acucciniello/alexa-open-doc#1]
I have no been able to pinpoint exactly when it happens but after some time period, my developer account stops sending an accessToken along with the Lambda request.  The work around for this is to open up the Alexa application on your phone and relink the skill.  This can pose a potential problem in the future for users to have to relink the skill every couple of hours for some reason.

**Comment by acucciniello** [comment]
It turns out Google Drive was giving me an authorization code not an access token.  The authorization code is temporary and it is used to get an access token.

**Comment by acucciniello** [comment]
I am now using authorization code grant instead of implicit grant, so google gives me an authorization code and then returns an access token.  The access_token that google gives me  expires after 1 hour.  Proposed Solution: to get a refresh token, so when the token does expire, code will automatically get me a new access_token, not causing any problems for the user

**Comment by acucciniello** [comment]
The last commit fixed this issue, by changing a setting in the Amazon Developer Portal.  The setting is titled "Client Authentication Scheme" the setting to make it work should be HTTP in order to properly get the refresh token from google.  Closing Issue

**Comment by acucciniello** [comment]
After adding CreateFileIntent and EditFileIntent refresh token seems to stop working, more investigating is needed to be done

**Comment by acucciniello** [comment]
After further investigation, it seems as if the refresh token is received only when calling the Launch Intent, versus calling any other intent that I have created.  I am not sure if this is something that happens with amazon or if there is a minute difference in the code that is causing a refresh token to not be received.  I cannot close this until I know for a fact that it will work without having to call the Launch Intent

**Comment by acucciniello** [comment]
After a combination of two things and some playing with the skill afterward, it seems as if the skill is automatically getting the refresh token as it is supposed to.  In my Amazon Developer Portal, under configuration, I have changed the setting of Client Authentication Scheme to `Credentials in request body`.  After that, I went into my Google Drive account settings, removed access to the skill, and then reallowed access by relinking the skill in the Amazon Alexa Companion App ( the mobile app).  My thought process is that by relinking the skill after it now had new capabilities, a new refresh token was needed.  That last step (of denying access then relinking) may have to occur when new google drive functionality is added to the skill in the future.  Closing this issue as resolved.
